pkg/divert/istio/divert.go

```diff
@@ -0,0 +1,222 @@
+	istioclientset "istio.io/client-go/pkg/clientset/versioned"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Driver weaver struct for the divert driver
+type Driver struct {
+	Client      kubernetes.Interface
+	IstioClient istioclientset.Interface
+	Manifest    *model.Manifest
+	cache       *divertIstioCache
+}
+
+type divertIstioCache struct {
+	divertVirtualServices    map[string]*istioV1beta1.VirtualService
+	developerVirtualServices map[string]*istioV1beta1.VirtualService
+}
+
+func (d *Driver) Deploy(ctx context.Context) error {
+	var err error
+	d.cache, err = initDivertIstioCache(ctx, d.Manifest, d.IstioClient)
+	if err != nil {
+		return err
+	}
+
+	for name := range d.cache.divertVirtualServices {
+		select {
+		case <-ctx.Done():
+			oktetoLog.Infof("deployDivert context cancelled")
+			return ctx.Err()
+		default:
+			oktetoLog.Spinner(fmt.Sprintf("Diverting virtual service %s/%s ...", d.Manifest.Deploy.Divert.Namespace, name))
+			if err := d.retryTranslateDivertVirtualService(ctx, name); err != nil {
+				return err
+			}
+
+			if !isInDivertVirtualServices(name, d.Manifest.Deploy.Divert.VirtualServices) {
+				continue
+			}
+			if err := d.retryTranslateDeveloperVirtualService(ctx, name); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (d *Driver) Destroy(ctx context.Context) error {
+	var err error
+	d.cache, err = initDivertIstioCache(ctx, d.Manifest, d.IstioClient)
+	if err != nil {
+		return err
+	}
+
+	for name := range d.cache.divertVirtualServices {
+		select {
+		case <-ctx.Done():
+			oktetoLog.Infof("destroyDivert context cancelled")
+			return ctx.Err()
+		default:
+			oktetoLog.Spinner(fmt.Sprintf("Destroying divert un virtual service %s/%s ...", d.Manifest.Deploy.Divert.Namespace, name))
+			if err := d.retryRestoreDivertVirtualService(ctx, name); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (d *Driver) ApplyNamespaceToProxy() string {
+	if d.Manifest.Deploy.Divert.Namespace == d.Manifest.Namespace {
+		return ""
+	}
+	return d.Manifest.Deploy.Divert.Namespace
+}
+
+func initDivertIstioCache(ctx context.Context, m *model.Manifest, c istioclientset.Interface) (*divertIstioCache, error) {
+	cache := &divertIstioCache{
+		divertVirtualServices:    map[string]*istioV1beta1.VirtualService{},
+		developerVirtualServices: map[string]*istioV1beta1.VirtualService{},
+	}
+	vsList, err := virtualservices.List(ctx, m.Deploy.Divert.Namespace, c)
+	if err != nil {
+		return nil, err
+	}
+	for i := range vsList {
+		cache.divertVirtualServices[vsList[i].Name] = vsList[i]
+	}
+
+	vsList, err = virtualservices.List(ctx, m.Namespace, c)
+	if err != nil {
+		return nil, err
+	}
+	for i := range vsList {
+		cache.developerVirtualServices[vsList[i].Name] = vsList[i]
+	}
+	return cache, nil
+}
+
+func (d *Driver) retryTranslateDeveloperVirtualService(ctx context.Context, name string) error {
+	retries := 0
+	var err error
+	for retries < 10 {
```

Comment:
Can we move the magic number to constants at the top of the file. We are using it also [in line 167](https://github.com/okteto/okteto/pull/3384/files#diff-0e4a458af0d49f8538599e45205648929cbd76be5470d3fabf89cb47756c3581R167)